app/ide-desktop/lib/client/src/authentication.ts

```diff
@@ -0,0 +1,221 @@
+    initIpc()
+    initOpenUrlListener(window)
+}
+
+/** Registers an Inter-Process Communication (IPC) channel between the Electron application and the
+ * served website.
+ *
+ * This channel listens for {@link ipc.channel.openUrlInSystemBrowser} events. When this kind of
+ * event is fired, this listener will assume that the first and only argument of the event is a URL.
+ * This listener will then attempt to open the URL in a cross-platform way. The intent is to open
+ * the URL in the system browser.
+ *
+ * This functionality is necessary because we don't want to run the OAuth flow in the app. Users
+ * don't trust Electron apps to handle their credentials. */
+const initIpc = () => {
+    electron.ipcMain.on(ipc.channel.openUrlInSystemBrowser, (_event, url) => opener(url))
+}
+
+/** Registers a listener that fires a callback for `open-url` events, when the URL is a deep link.
+ *
+ * This listener is used to open a page in *this* application window, when the user is
+ * redirected to a URL with a protocol supported by this application.
+ *
+ * All URLs that aren't deep links (i.e., URLs that don't use the {@link shared.DEEP_LINK_SCHEME}
+ * protocol) will be ignored by this handler. Non-deep link URLs will be handled by Electron. */
+const initOpenUrlListener = (window: () => electron.BrowserWindow) => {
+    electron.app.on(OPEN_URL_EVENT, (event, url) => {
+        const parsedUrl = new URL(url)
+        /** Prevent Electron from handling the URL at all, because redirects can be dangerous. */
+        event.preventDefault()
+        if (parsedUrl.protocol !== `${shared.DEEP_LINK_SCHEME}:`) {
+            content.logger.error(`${url} is not a deep link, ignoring.`)
+            return
+        }
+        window().webContents.send(ipc.channel.openDeepLink, url)
+    })
+}
+
+
+
+// ==============
+// === Opener ===
+// ==============
+
+
+/** Function for opening URLs in the system browser in a cross-platform way.
+ *
+ * This function contains the contents of the `opener` NPM package, as of commit
+ * https://github.com/domenic/opener/commit/24edf48a38d1e23bbc5ffbeb079c206d5565f062.  To avoid an
+ * extra external dependency, the contents of the function have been copied into this file. The
+ * function has been modified to follow the Enso style guide, and to be TypeScript-compatible. */
+const opener = (args: any, options?: any, callback?: any) => {
+    let platform = process.platform
+
+    /** Attempt to detect Windows Subystem for Linux (WSL). WSL  itself as Linux (which works in
+     * most cases), but in this specific case we need to treat it as actually being Windows. The
+     * "Windows-way" of opening things through cmd.exe works just fine here, whereas using xdg-open
+     * does not, since there is no X Windows in WSL. */
+    if (platform === 'linux' && os.release().indexOf('Microsoft') !== -1) {
+        platform = 'win32'
+    }
+
+    /** http://stackoverflow.com/q/1480971/3191, but see below for Windows. */
```

Comment:
Wait, we've been talking about that. I asked why this dependency, you explained that it is small platform-specific implementation, and I thought that we were on the same page. Copy-pasting code of it is worse then depending on the dependency I think.

Reply:
Fixed by moving the opener dependency back to a dependency.